src/tools/manifest-checker.ts

```diff
@@ -0,0 +1,74 @@
+/**
+ * @license
+ * Copyright (c) 2019 Google Inc. All rights reserved.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * Code distributed by Google as part of this project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+import minimist from 'minimist';
+import {Manifest, ManifestWarning} from '../runtime/manifest.js';
+import {PlatformLoader} from '../platform/loader-web.js';
+
+// Script to check that a bundle of Arcs manifest files, particle
+// implementations and JSON data files is complete (i.e. no explicitly mentioned
+// dependencies are missing).
+//
+// To be run with the arcs_manifest BUILD rule.
+
+/**
+ * Loads the given .arcs manifest file and checks it for errors. Errors are
+ * thrown as an exception.
+ */
+async function checkManifest(src: string) {
+  const loader = new PlatformLoader({});
+  const manifest = await Manifest.load(src, loader);
+
+  // Look for errors from parsing the manifest (ignore warnings). This covers
+  // missing .arcs imports.
+  const manifestErrors = manifest.errors
+      .filter(error => !(error instanceof ManifestWarning))
+      .map(error => error.toString());
+
+  if (manifestErrors.length) {
+    throw manifestErrors.join('\n');
+  }
+
+  // Check particle impls can be loaded.
+  for (const particle of manifest.particles) {
+    const implFile = particle.implFile;
+    if (implFile.endsWith('.wasm')) {
```

Comment:
Is this backwards? Should that be if !implFile.endsWith('.wasm') ?

Reply:
Good catch, thanks